upup/pkg/fi/cloudup/awstasks/launchtemplate.go

```diff
@@ -154,22 +148,17 @@ func (t *LaunchTemplate) CheckChanges(a, e, changes *LaunchTemplate) error {
 func (t *LaunchTemplate) FindDeletions(c *fi.Context) ([]fi.Deletion, error) {
 	var removals []fi.Deletion
 
-	configurations, err := t.findLaunchTemplates(c)
+	configurations, err := t.findAllLaunchTemplates(c)
 	if err != nil {
 		return nil, err
 	}
 
-	if len(configurations) <= RetainLaunchConfigurationCount() {
-		return nil, nil
-	}
-
-	configurations = configurations[:len(configurations)-RetainLaunchConfigurationCount()]
-
+	prefix := fmt.Sprintf("%s-", fi.StringValue(t.Name))
 	for _, configuration := range configurations {
-		removals = append(removals, &deleteLaunchTemplate{lc: configuration})
+		if strings.HasPrefix(aws.StringValue(configuration.LaunchTemplateName), prefix) {
```

Comment:
Why is this necessary? Wouldn't all the old launch templates be deleted as owned by the cluster but unreferenced by an ASG?

Reply:
There can be N previous LTs for each ASG. already unreferenced that should be deleted.
Maybe there's some mechanism that I'm missing.

Reply:
Looks like I misunderstood the `Find` / `FindDeletions` mechanisms. I thought it would look for resources tagged as owned by the cluster and delete them if they're not expected.